File: versão em python/3.py
# 3 - Questão:

import random


class Craps:
    def __init__(self):
        self.jogada = 0
        self.ponto = 0

    def menu(self):
        opcao = -1
        while opcao != 0:
            print()
            self.mostrar_ponto()
            print('- - - Selecione uma opção: - - -')
            print('1 - Lançar dados')
            print('0 - Sair')
            opcao = input('Digite a opção: ')

            if opcao == '1':
                self.mostrar_resultado(self.lancar_dados())
            elif opcao == '0':
                break
            else:
                print('\nSelecione uma opção válida!')

    def mostrar_ponto(self):
        if self.ponto != 0:
            print("Seu ponto é:", self.ponto)

    def mostrar_resultado(self, resultado):
        print("\n======================================")
        print("Jogada:", self.jogada)
        print("Resultado =", resultado)
        if self.jogada == 1:
            if resultado in [7, 11]:
                print("\n'Natural', você ganhou!")
                self.resetar()
            elif resultado in [2, 3, 12]:
                print("\n'Craps', você perdeu!")
                self.resetar()
            else:
                self.ponto = resultado
                print()
                print(self.ponto, "é o seu ponto!")
                print("Continue a lançar os dados até tirar este ponto novamente!")
        else:
            if resultado == self.ponto:
                print("\nVocê Ganhou!")
                self.resetar()
            elif resultado == 7:
                print("\nVocê Perdeu!")
                self.resetar()
        print("======================================")

    def lancar_dados(self):
        self.jogada += 1
        resultado = random.randint(2, 12)
        return resultado

    def resetar(self):
        self.jogada = 0
        self.ponto = 0


if __name__ == "__main__":
    main = Craps()
    main.menu()

File: 3.rb
# 3 - Questão:


class Craps
    @@jogada = 0
    @@ponto = 0

    def menu
        opcao = -1
        while opcao != 0
            puts ""
            mostrar_ponto
            puts "- - - Selecione uma opção: - - -"
            puts "1 - Lançar dados"
            puts "0 - Sair"

            print "Digite a opção: "
            opcao = gets.chomp

            if opcao == "1"
                mostrar_resultado(lancar_dados)
            elsif opcao == "0"
                break
            else
                puts "\nSelecione uma opção válida!"
            end
        end
    end

    def lancar_dados
        @@jogada += 1
        resultado = rand(2..12)
        return resultado
    end

    def mostrar_resultado(resultado)
        puts "\n======================================"
        puts "Jogada: #{@@jogada}"
        puts "Resultado = #{resultado}"
        if @@jogada == 1
            if [7, 11].include?(resultado)
                puts "\n'Natural', você ganhou!"
                resetar
            elsif [2, 3, 12].include?(resultado)
                puts "\n'Craps', você perdeu!"
                resetar
            else
                @@ponto = resultado
                puts ""
                puts "#{@@ponto} é o seu ponto!"
                puts "Continue a lançar os dados até tirar este ponto novamente!"
            end
        else
            if resultado == @@ponto
                puts "\nVocê Ganhou!"
                resetar
            elsif resultado == 7
                puts "\nVocê Perdeu!"
                resetar
            end
        end
        puts "======================================"
    end

    def mostrar_ponto
        if @@ponto != 0
            puts "Seu ponto é: #{@@ponto}"
        end
    end

    def resetar
        @@jogada = 0
        @@ponto = 0
    end
end


main = Craps.new
main.menu
